static/coffee/controllers.coffee

```diff
@@ -270,7 +270,17 @@ controllers.controller('MessagesController', ['$scope', '$timeout', '$uibModal',
     )
 
   $scope.fetchOldItems = (search, startTime, page) ->
-    return MessageService.fetchOld(search, startTime, page)
+    $scope.autoRefresh = ->
+      MessageService.fetchOld(search, startTime, page).then((data) ->
+        $scope.items = data.results
+        $scope.oldItemsMore = data.hasMore
+        $scope.oldItemsLoading = false
+      )
+    $scope.autoRefresh()
+    $interval($scope.autoRefresh, 30000)
```

Comment:
Should we add an `$interval.cancel()` above this line incase `fetchOldItems()` is called more than once for some reason?